### ISSUE TITLE
Move version string to interface layer

### ISSUE THREAD

**Issue by roigcarlo** [KratosMultiphysics/Kratos#10865]
The version string for printing the python minor in the greeting message is creating a (very dark, hidden and nasty) compile time dependency on the kernel, which is preventing the re-usability of the kernel while targeting different interface version.

Mostly the problem is in `kratosCore.dir/flags.make`:
```make
flags.make:CXX_DEFINES = -DBOOST_UBLAS_MOVE_SEMANTICS -DKRATOS_CORE=IMPORT,API -DKRATOS_MAJOR_VERSION=9 -DKRATOS_MINOR_VERSION=3 -DKRATOS_PYTHON -DKRATOS_SMP_OPENMP -DKRATOS_USING_MPI -DKratosCore_EXPORTS -DPYTHON_VERSION_MAJOR=3 -DPYTHON_VERSION_MINOR=11 -DPYTHON_VERSION_PATCH=0 -DUSE_TRIANGLE_NONFREE_TPL
```

In order to prevent this we should initialize the python version AFTER the kernel has been loaded and otherwise assign a default in the kernel.

This is important as it reduces the current wheel release build time to 1/4.

**Comment by loumalouomega** [comment]
The easiest thing I can think is to define is a dynamic library of these static values, so if changed is just a matter of relink.

**Comment by roigcarlo** [comment]
Don't worry, is solved at #10867

Actually the problem was the mechanism we use to pass the python version to the kernel, not the fact that it knows it by itself. We were adding `-DPYTHON_VERSION_MAJOR=XXX` as a flag to the whole project rather than adding it only to the interface.

Once this was solved it was just a matter of storing the version in the kernel and populate it when it is loaded rather than when it is compiled.
